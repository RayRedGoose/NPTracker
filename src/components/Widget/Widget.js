import './Widget.scss'
import React from 'react'
import Map from '../Map/Map'

const Widget = ({title, text, data}) => {
  return (
    <section className={`widgets ${title}`}>
      <h3>{ title } info</h3>
      {
        (text)
          ? <p>{ text }</p>
          : <p>There is no { title } info.</p>
      }
      { title === 'directions' && <Map {...data}/> }
    </section>
  )
}

export default Widget
