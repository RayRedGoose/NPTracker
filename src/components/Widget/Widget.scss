@import "../../variables.scss";

.widgets {
  padding: 20px;
  h3 {
    color: $accent-color-primary-dark;
    font-family: Salsa;
    text-align: center;
    text-transform: capitalize;
    margin-bottom: 20px;
  }
  p {
    font-size: 14px;
    line-height: 1.5em;
  }
  &.directions {
    grid-area: direct;
    height: 100%;
    &>p {
      margin-bottom: 20px;
      height: 240px;
      overflow: scroll;
    }
    .place {
      color: black;
      font-size: 12px;
      font-weight: 700;
      width: 50%;
    }
  }
  &.weather {
    grid-area: weather;
  }
}
