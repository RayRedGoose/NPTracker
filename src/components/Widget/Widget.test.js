import React from 'react'
import { shallow } from 'enzyme'
import Widget from './Widget'

describe('Widget', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const widget = shallow(
      <Widget title="weather" text="Weather info text" />
    )

    expect(widget).toMatchSnapshot()
  })

  it('should match the snapshot if there is no text', () => {
    const widget = shallow(
      <Widget title="weather" />
    )

    expect(widget).toMatchSnapshot()
  })

  it('should match the snapshot if it\'s directions widget', () => {
    const widget = shallow(
      <Widget
        title="directions"
        text="This is directions info text"
        data={{name: 'Mount Rainier'}} />
    )

    expect(widget).toMatchSnapshot()
  })
})
