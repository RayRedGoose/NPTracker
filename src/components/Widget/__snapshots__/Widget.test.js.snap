// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget should match the snapshot if it's directions widget 1`] = `
<section
  className="widgets directions"
>
  <h3>
    directions
     info
  </h3>
  <p>
    This is directions info text
  </p>
  <Map
    name="Mount Rainier"
  />
</section>
`;

exports[`Widget should match the snapshot if there is no text 1`] = `
<section
  className="widgets weather"
>
  <h3>
    weather
     info
  </h3>
  <p>
    There is no 
    weather
     info.
  </p>
</section>
`;

exports[`Widget should match the snapshot with all data passed in correctly 1`] = `
<section
  className="widgets weather"
>
  <h3>
    weather
     info
  </h3>
  <p>
    Weather info text
  </p>
</section>
`;
